### PR TITLE
Adding manifest generation to build

### PR DESCRIPTION
**Pull request by scbedd**
Related to Azure/azure-sdk-tools#2273

Thanks @seankane-msft and @richardpark-msft for the help here!

### CODE REVIEW

File: eng/pipelines/templates/steps/build-test.yml [hunk 1]
```diff
@@ -66,6 +66,15 @@ steps:
...
       condition: succeededOrFailed()
 
+  - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
+    displayName: 'Generate BOM'
+    inputs:
+      BuildDropPath: ${{ parameters.ServiceDirectory }}
+
+  - publish: ${{ parameters.ServiceDirectory }}/_manifest
+    displayName: 'Publish BOM Manifest'
+    artifact: 'manifest'
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-test.yml:76]: Should we be consistent and call all the uploads manifest for the other languages as well?

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/steps/build-test.yml (hunk 1 above)
Comment (scbedd) [reply eng/pipelines/templates/steps/build-test.yml:76]: Yessir. I'm actually adjusting these in the original PRs to match.

File: eng/pipelines/templates/jobs/archetype-sdk-client-samples.yml [hunk 2]
```diff
@@ -14,18 +14,22 @@ stages:
             pool.name: azsdk-pool-mms-ubuntu-2004-general
             image.name: MMSUbuntu20.04
             go.version: '1.17'
+            generate.bom: false
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client-samples.yml:17]: Given this is the default value is it worth copying everywhere?

Comment (scbedd) [reply eng/pipelines/templates/jobs/archetype-sdk-client-samples.yml:17]: You have a good point. I'll try it.

**Pull request merged**